autoscheduler/frontend/src/components/SchedulingPage/Schedule/Schedule.tsx

```diff
@@ -399,6 +420,47 @@ const Schedule: React.FC = () => {
     </div>
   ));
 
+  // When term is chagned, fetch saved availabilities for the new term
+  React.useEffect(() => {
+    if (term) {
+      fetch(`sessions/get_saved_availabilities?term=${term}`).then(
+        (res) => res.json(),
+      ).then((avails: Availability[]) => {
+        // We're done loading - hide the loading indicator and set the new availabilities
+        dispatch(setAvailabilities(avails));
+        setIsLoadingAvailabilities(false);
+      });
+    }
+
+    // on unmount, clear availabilities
+    return (): void => {
+      dispatch(clearAvailabilities());
+    };
+  }, [term, dispatch]);
+
+  // Whenever we're not clicking, save availabilities every 15 seconds
+  React.useEffect(() => {
+    if (!term) return;
+
+    // Only call throttle once we've stopped dragging (and thus stopped making changes) and
+    // when availabilities are still loading
+    if (isMouseDown || isLoadingAvailabilities) return;
+
+    // Serialize availabilities and make API call
+    const saveAvailabilities = (): void => {
+      fetch('sessions/save_availabilities', {
+        method: 'PUT',
+        headers: {
+          'Content-Type': 'application/json',
+          'X-CSRFToken': Cookies.get('csrftoken'),
+        },
+        body: JSON.stringify({ term, availabilities: availabilityList }),
+      });
+    };
+
+    throttle(`${term}`, saveAvailabilities, 15000, true);
```

Comment:
```suggestion
    const saveAllAvailabilities = (): void => { throttle('', () => {}, 2 ** 31 - 1, true); };
    if (!term) return saveAllAvailabilities;

    // Only call throttle once we've stopped dragging (and thus stopped making changes) and
    // when availabilities are still loading
    if (isMouseDown || isLoadingAvailabilities) return saveAllAvailabilities;

    // Serialize availabilities and make API call
    const saveAvailabilities = (): void => {
      fetch('sessions/save_availabilities', {
        method: 'PUT',
        headers: {
          'Content-Type': 'application/json',
          'X-CSRFToken': Cookies.get('csrftoken'),
        },
        body: JSON.stringify({ term, availabilities: availabilityList }),
      });
    };

    throttle(`${term}`, saveAvailabilities, 15000, true);
    return saveAllAvailabilities;
```
Saving courses has the same problem as this where if you go back to the homepage then select the same term again, it will load your old availabilities (since they don't get saved when you go back to the homepage). We need to save on unmount and you can do it like this, I noticed this bug yesterday and tried making it for saving courses, but it broke some tests and I didn't want to deal with it then. This doesn't break any tests though so might as well make it in this branch (and also for saving schedules)